### PR TITLE
server: implement workspace/didChangeWatchedFiles

### PR DESCRIPTION
**Pull request by nedpals**
According to [LSP 3.15 Specification:](https://microsoft.github.io/language-server-protocol/specifications/specification-3-15/#workspace_didChangeWatchedFiles)
>
>The watched files notification is sent from the client to the server when the client detects changes to files watched by the language client. It is recommended that servers register for these file events using the registration mechanism. In former implementations clients pushed file events without the server actively asking for it.
>
>Servers are allowed to run their own file watching mechanism and not rely on clients to provide file events. However this is not recommended due to the following reasons:
>
> - to our experience getting file watching on disk right is challenging, especially if it needs to be supported across multiple OSes.
> - file watching is not for free especially if the implementation uses some sort of polling and keeps a file tree in memory to compare time stamps (as for example some node modules do)
> - a client usually starts more than one server. If every server runs its own file watching it can become a CPU or memory problem.
> - in general there are more server than client implementations. So this problem is better solved on the client side.

**TL;DR**: This is an important feature in order to handle file deletion and file renaming for files not currently opened in the editor. 

## Notes
See https://github.com/vlang/vls/pull/228/files#r712674414

### CODE REVIEW

File: server/workspace.v [hunk 1]
```diff
@@ -1,8 +1,125 @@
...
+	mut is_rename := false
+	
+	// NOTE:
+	// 1. Renaming a file returns two events: one "deleted" event for 
+	//    the file with old name and one "created" event for the same
+	//    file with new name.
+	// 2. Deleting a folder does not trigger a "deleted" event. Restoring
+	//    the files of the folder however triggers the "created" event.
+	// 3. Renaming a folder triggers the "created" event for each file
+	//    but with no "deleted" event prior to it.
```
Comment (nedpals) [inline server/workspace.v:23]: .
